drivers/sensor/lsm6dso16is/lsm6dso16is_shub.c

```diff
@@ -0,0 +1,838 @@
+/* ST Microelectronics LSM6DSO16IS 6-axis IMU sensor driver
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lsm6dso16is.pdf
+ */
+
+#define DT_DRV_COMPAT st_lsm6dso16is
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+
+#include "lsm6dso16is.h"
+
+LOG_MODULE_DECLARE(LSM6DSO16IS, CONFIG_SENSOR_LOG_LEVEL);
+
+#define LSM6DSO16IS_SHUB_DATA_OUT				0x02
+
+#define LSM6DSO16IS_SHUB_SLV0_ADDR				0x15
+#define LSM6DSO16IS_SHUB_SLV0_SUBADDR			0x16
+#define LSM6DSO16IS_SHUB_SLV0_CONFIG			0x17
+#define LSM6DSO16IS_SHUB_SLV1_ADDR				0x18
+#define LSM6DSO16IS_SHUB_SLV1_SUBADDR			0x19
+#define LSM6DSO16IS_SHUB_SLV1_CONFIG			0x1A
+#define LSM6DSO16IS_SHUB_SLV2_ADDR				0x1B
+#define LSM6DSO16IS_SHUB_SLV2_SUBADDR			0x1C
+#define LSM6DSO16IS_SHUB_SLV2_CONFIG			0x1D
+#define LSM6DSO16IS_SHUB_SLV3_ADDR				0x1E
+#define LSM6DSO16IS_SHUB_SLV3_SUBADDR			0x1F
+#define LSM6DSO16IS_SHUB_SLV3_CONFIG			0x20
+#define LSM6DSO16IS_SHUB_SLV0_DATAWRITE			0x21
```

Comment:
Replace "SLV" too

Reply:
I could delete those names and use the ones defined in https://github.com/zephyrproject-rtos/hal_st/blob/master/sensor/stmemsc/lsm6dsv16x_STdC/driver/lsm6dsv16x_reg.h, like LSM6DSV16X_SLV0_ADD. Still the SLV keyword in it, but it's a h/w register name. I'll do the change anyway, since it is cleaner...

Reply:
If these are the documented register names it probably doesn't make sense to change. This is in a .c file not in a dts binding, kconfig, or header than an application may directly interact with. So probably ok in my mind to leave as is, but would be nice if st similarly updated their naming scheme.

Reply:
> If these are the documented register names it probably doesn't make sense to change. It would make it difficult to correlate the defines to the docs. This is in a .c file not in a dts binding, kconfig, or header than an application may directly interact with. So probably ok in my mind to leave as is, but would be nice if st similarly updated their naming scheme.

Yes, I will keep using the h/w reg names. Still, a change here is requested as it would be better to use directly the names defined in STMEMSC HAL and not redefine them. And yes, STM is going to uniform the names for the future devices (not the current/past ones) with the inclusive naming schemes.

Reply:
For the rest of names, I will replace ```slave``` with ```target``` (in all the possible declinations). Thanks!